pythonforandroid/bootstraps/common/build/src/main/java/org/kivy/android/PythonUtil.java

```diff
@@ -170,7 +170,74 @@ public static void unpackData(
             target.mkdirs();
 
             AssetExtract ae = new AssetExtract(ctx);
-            if (!ae.extractTar(resource + ".mp3", target.getAbsolutePath())) {
+            if (!ae.extractTar(resource + ".tar", target.getAbsolutePath(), "private")) {
+                String msg = "Could not extract " + resource + " data.";
+                if (ctx instanceof Activity) {
+                    toastError((Activity)ctx, msg);
+                } else {
+                    Log.v(TAG, msg);
+                }
+            }
+
+            try {
+                // Write .nomedia.
+                new File(target, ".nomedia").createNewFile();
+
+                // Write version file.
+                FileOutputStream os = new FileOutputStream(diskVersionFn);
+                os.write(dataVersion.getBytes());
+                os.close();
+            } catch (Exception e) {
+                Log.w("python", e);
+            }
+        }
+    }
+
+    public static void unpackPyBundle(
+        Context ctx,
+        final String resource,
+        File target,
+        boolean cleanup_on_version_update) {
+
+        Log.v(TAG, "Unpacking " + resource + " " + target.getName());
+
+        // The version of data in memory and on disk.
+        String dataVersion = "p4aisawesome"; // FIXME: Assets method is not usable for fake .so files bundled as a library.
+        String diskVersion = null;
+
+        Log.v(TAG, "Data version is " + dataVersion);
+
+        // If no version, no unpacking is necessary.
+        if (dataVersion == null) {
```

Comment:
How can this be null if it was set above?

Reply:
Removed